language-plutus-core/test/Evaluation/DynamicBuiltins/Definition.hs

```diff
@@ -49,26 +54,27 @@ test_dynamicFactorial :: TestTree
 test_dynamicFactorial =
     testCase "dynamicFactorial" $ do
         let env = insertDynamicBuiltinNameDefinition dynamicFactorialDefinition mempty
-            lhs = typecheckEvaluateCek env $ Apply () dynamicFactorial (makeIntConstant 10)
-            rhs = typecheckEvaluateCek mempty $ Apply () factorial (makeIntConstant 10)
+            ten = mkConstant @Integer @DefaultUni () 10
```

Comment:
Does the original definition have a `forall` to make the order explicit? Might be worth it.

Reply:
`mkConstant` is a new definition and it does have an explicit `forall` (we rarely need to specify a universe, most of the time we only need to specify a type, hence the order).